app/Rules/Valid.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace App\Rules;
+
+use Closure;
+
+use Illuminate\Support\Str;
+use Illuminate\Validation\ValidationException;
+use Illuminate\Support\Facades\Validator as ValidatorFacade;
+
+/**
+ * Rule generator
+ * default:
+ * - value is present and is of specified type
+ * sometimes:
+ * - if value doesn't appear in data, other validations (also nullable) are skipped
+ * nullable:
+ * - if value is null, other validations are skipped, [] and " " are not null in opposite to Laravel validation
+ */
+class Valid extends AbstractDataRule
+{
+    public static function bool(array $rules = [], bool $sometimes = false, bool $nullable = false): array
+    {
+        return self::base($sometimes, $nullable, ['boolean', DataTypeRule::bool()], $rules);
+    }
+
+    public static function int(array $rules = [], bool $sometimes = false, bool $nullable = false): array
+    {
+        return self::base($sometimes, $nullable, ['numeric', 'integer', DataTypeRule::int()], $rules);
+    }
+
+    public static function string(
+        array $rules = [],
+        bool $sometimes = false,
+        bool $nullable = false,
+        int $max = 250,
+    ): array {
+        return self::base($sometimes, $nullable, ['string', 'min:1', "max:$max"], $rules);
+    }
+
+    public static function trimmed(
+        array $rules = [],
+        bool $sometimes = false,
+        bool $nullable = false,
+        int $max = 250,
+    ): array {
+        return self::base($sometimes, $nullable, ['string', 'min:1', "max:$max", new StringIsTrimmedRule()], $rules);
+    }
+
+    public static function uuid(
+        array $rules = [],
+        bool $sometimes = false,
+        bool $nullable = false,
+    ): array {
+        return self::base($sometimes, $nullable, ['string', 'lowercase', 'uuid'], $rules);
+    }
+
+    public static function array(
```

Comment:
Zmieniłbym tę nazwę na coś bardziej opisowego typu "hasKeys".

Reply:
Zostawiłbym nazwę array, bo to jest odwołanie do tego czym jest walidacja array wbudowana w laravela. Czyli mogę chcieć jej użyć bez parametru keys, że ma przyjąć dowolną tablicę

Reply:
I nie has, tylko canHave